src/persist-types/src/codec_impls.rs

```diff
@@ -362,6 +397,35 @@ impl Codec for Vec<u8> {
     }
 }
 
+impl Codec for Bytes {
+    type Storage = ();
+    type Schema = VecU8Schema;
+
+    fn codec_name() -> String {
+        "Bytes".into()
```

Comment:
If someone tries to open an existing shard that was written with the `Vec<u8>` codec, it will error with an invalid codec error. This is maybe unfortunate because the two codecs should be compatible but you can't migrate existing `Vec<u8>` shards to `Bytes`.

A hack I did in my staging env is to name this Codec `"Vec<u8>".into()` to trick persist into thinking they're the same codec.

Reply:
Alternatively, for any shard that wants to switch from `"Vec<u8>"` to `"Bytes"` would have to migrate to a new shard with the new codec.

Reply:
Since both "Vec<u8>" and "Bytes" are both byte blobs, I wouldn't be opposed to calling them the same thing?

Reply:
@bkirwi any thoughts here? IMO using `"Vec<u8>"` as the codec name is okay since fundamentally both types are just blobs of bytes

Reply:
I don't feel strongly about it!